### PR TITLE
contributing: add commit message and pr style

### PR DESCRIPTION
**Pull request by lilin90**
@shenli PTAL

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -268,21 +268,17 @@ Please follow this style to make TiDB easy to review, maintain and develop.
...
-[Os X GNU tools]: https://www.topbug.net/blog/2013/04/14/install-and-use-gnu-command-line-tools-in-mac-os-x
-[go-1.8]: https://blog.golang.org/go1.8
-[go-workspace]: https://golang.org/doc/code.html#Workspaces
-[issue]: https://github.com/pingcap/tidb/issues
-[mercurial]: http://mercurial.selenic.com/wiki/Download
+- For the first subject line, use no more than 70 characters.
+- For the second line, always leave it blank.
+- For other lines, use no more than 80 characters.
+- If the change affects two subsystems, you can use a comma (and a whitespace) to separate them like `util/codec, util/types:`.
+- If the changes affects three or more subsystems, you can use `*` instead, like `*:`.
```
Comment (shenli) [inline CONTRIBUTING.md]: s/affects/affect/

File: commit-message-pr-style.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+<why this change was made>
+<BLANK LINE>
+<footer>(optional)
+```
+
+- For the first subject line, use no more than 70 characters.
+- For the second line, always leave it blank.
+- For other lines, use no more than 80 characters.
+- If the change affects two subsystems, you can use a comma (and a whitespace) to separate them like `util/codec, util/types:`.
+- If the changes affects three or more subsystems, you can use `*` instead, like `*:`.
```
Comment (shenli) [inline commit-message-pr-style.md]: s/affects/affect/

File: commit-message-pr-style.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+```
+<subsystem>: <what changed>
+<BLANK LINE>
+<why this change was made>
+<BLANK LINE>
+<footer>(optional)
+```
+
+- For the first subject line, use no more than 70 characters.
```
Comment (shenli) [inline commit-message-pr-style.md]: Is this part duplicated with the content in CONTRIBUTING.md?

File: commit-message-pr-style.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+- If the change affects two subsystems, you can use a comma (and a whitespace) to separate them like `util/codec, util/types:`.
+- If the changes affects three or more subsystems, you can use `*` instead, like `*:`.
+- For the why part, if there is no specific reason for the change, you can use one of the generic reasons like "Improve performance", "Improve test coverage."
+
+### Habits for a good commit message
+
+- Summarize your change
+- Describe clearly one logical change and avoid lazy messages as `misc fixes`
+- Describe any limitations of the current code
+- Do not end the subject with a period “.”
```
Comment (shenli) [inline commit-message-pr-style.md]: “.” -> "."

File: commit-message-pr-style.md (hunk 3 above)
Comment (lilin90) [reply commit-message-pr-style.md]: @shenli Yes. This part is the same as that in CONTRIBUTING.md. This document is a more detailed version. Could we keep them both?

**Comment by shenli** [comment]
LGTM

**Pull request merged**